cursive-core/src/views/dialog.rs

```diff
@@ -613,69 +644,148 @@ impl Dialog {
+                if printer.size.y < overhead.horizontal() {
+                    return None;
+                }
+                let mut offset = overhead.top
+                    + self
+                        .align
+                        .v
+                        .get_offset(height, printer.size.y - overhead.vertical());
+
+                let overhead_right = self.padding.right + self.borders.right + max_size;
+
+                let x = match printer.size.x.checked_sub(overhead_right) {
+                    Some(x) => x,
+                    None => return None,
+                };
+
+                for (i, button) in self.buttons.iter().enumerate() {
+                    let size = button.button.size;
+                    // Add some special effect to the focused button
+                    let position = Vec2::new(x, offset);
+                    *button.offset.lock() = position;
+                    button.button.draw(
+                        &printer
+                            .offset(position)
+                            .cropped(size)
+                            .focused(self.focus == DialogFocus::Button(i)),
+                    );
+                    // Keep 1 blank between two buttons
+                    offset += size.y + 1;
+                    // Also keep 1 blank to the left of the buttons
+                    buttons_width = max(buttons_width, size.x + 1);
+                }
 
-        for (i, button) in self.buttons.iter().enumerate() {
-            let size = button.button.size;
-            // Add some special effect to the focused button
-            let position = Vec2::new(offset, y);
-            *button.offset.lock() = position;
-            button.button.draw(
-                &printer
-                    .offset(position)
-                    .cropped(size)
-                    .focused(self.focus == DialogFocus::Button(i)),
-            );
-            // Keep 1 blank between two buttons
-            offset += size.x + 1;
-            // Also keep 1 blank above the buttons
-            buttons_height = max(buttons_height, size.y + 1);
+                return Some(buttons_width);
+            }
         }
-
-        Some(buttons_height)
     }
 
-    fn draw_content(&self, printer: &Printer, buttons_height: usize) {
+    fn draw_content(&self, printer: &Printer, buttons_size: usize) {
         // What do we have left?
-        let taken =
-            Vec2::new(0, buttons_height) + self.borders.combined() + self.padding.combined();
-
-        let inner_size = match printer.size.checked_sub(taken) {
-            Some(s) => s,
-            None => return,
-        };
-
-        self.content.draw(
-            &printer
-                .offset(self.borders.top_left() + self.padding.top_left())
-                .cropped(inner_size)
-                .focused(self.focus == DialogFocus::Content),
-        );
+        match self.vertical_button_orientation {
```

Comment:
Here too we should be able to share the implementation.

Either directly take `buttons_size` as a `Vec2` in the function, or:
```rust
let buttons_size = Vec2::from_major_minor(self.buttons_orientation, buttons_size, 0)`
```

Then you can do

```rust
let taken = buttons_size + self.borders.combined() + self.padding.combined();
```

And continue with the shared logic.

Reply:
I managed to turn it into a single implementation with the advice you provided.